pkg/kube/dialer_test.go

```diff
@@ -0,0 +1,132 @@
+package kube
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go/modules/k3s"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestDialer(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	container, err := k3s.Run(ctx, "rancher/k3s:v1.27.1-k3s1")
```

Comment:
Let's chat about testcontainer's `k3s` vs `k3d` vs `kind` soon and come up with some guidelines. We now have 3 competing standards. Alternatively, if the k3s container starts up in less than 30s I may be sold.